submissions/Levi123/oop-exercise/index.js

```diff
@@ -0,0 +1,112 @@
+// class Animal {
+//    constructor (species, name, gender){
+//       this.species = species;
+//       this.name = name;
+//       this.gender = gender;
+//    }
+// }
+
+class Entity {
+   constructor (options){
```

Comment:
Code is written for other people.
As a peer developer I am tasked to add a class that extends entity. In our team we are used to rely on code editors intellisense. Having current constructor specification it will only tell that constructor expects some "options". We would need to go through the code and read the method body to understand what is expected. As a team member you will also love when peer developers take care of you and write clear code.

At the same time if function specification is detailed there is no such need. There are methods to write self-explaining functions specifications: (1) regular list of parameters, (2) object destructuring syntax when parameters are supplied as objects. Use either of those.

Reply:
> Code is written for other people. As a peer developer I am tasked to add a class that extends entity. In our team we are used to rely on code editors intellisense. Having current constructor specification it will only tell that constructor expects some "options". We would need to go through the code and read the method body to understand what is expected. As a team member you will also love when peer developers take care of you and write clear code.
> 
> At the same time if function specification is detailed there is no such need. There are methods to write self-explaining functions specifications: (1) regular list of parameters, (2) object destructuring syntax when parameters are supplied as objects. Use either of those.

Object destructuring, I understood correctly?

For example:
class Entity {
   constructor ({name, species, gender, saying){
      this.name = name;
      this.species = species;
      this.gender = gender;
      this.saying = saying;
   }
}